cloudevents/pydantic/v2/conversion.py

```diff
@@ -0,0 +1,75 @@
+#  Copyright 2018-Present The CloudEvents Authors
```

Comment:
All these functions are the same as the V1 ones, I thought about using the same V1/V2 detection logic as the __init__.py file.

If we want to keep the ability for pydantic V2 users to choose the V1 implementation anyway, we need to keep them separate and duplicate.